src/main/java/seedu/address/model/person/Telegram.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's telegram in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTelegram(String)}
+ */
+public class Telegram {
+
+    public static final String MESSAGE_CONSTRAINTS = "Telegram handle can take any values, and it should start with @";
+
+    /*
+     * The first character of the telegram handle should be @,
+     * blank string should not be a valid input.
+     */
+    public static final String VALIDATION_REGEX = "@[^\\s].*";
```

Comment:
for user friendliness, would it be possible to append the @ automatically if not included?

Reply:
this should also possibly check for no spaces in the handle

Reply:
I think the professor should key in the @ manually so that they know it is a telegram handle (same logic for email)